src/irrlicht_changes/CGUITTFont.cpp

```diff
@@ -403,19 +418,14 @@ CGUITTGlyphPage* CGUITTFont::createGlyphPage(const u8 pixel_mode)
 	page = new CGUITTGlyphPage(Driver, name);
 
 	// Determine our maximum texture size.
-	// If we keep getting 0, set it to 1024x1024, as that number is pretty safe.
-	core::dimension2du max_texture_size = max_page_texture_size;
-	if (max_texture_size.Width == 0 || max_texture_size.Height == 0)
-		max_texture_size = Driver->getMaxTextureSize();
-	if (max_texture_size.Width == 0 || max_texture_size.Height == 0)
-		max_texture_size = core::dimension2du(1024, 1024);
-
-	// We want to try to put at least 144 glyphs on a single texture.
+	core::dimension2du max_texture_size = Driver->getMaxTextureSize();
+
+	// We want to try to put at least 180 glyphs on a single texture.
 	core::dimension2du page_texture_size;
-	if (size <= 21) page_texture_size = core::dimension2du(256, 256);
-	else if (size <= 42) page_texture_size = core::dimension2du(512, 512);
-	else if (size <= 84) page_texture_size = core::dimension2du(1024, 1024);
-	else if (size <= 168) page_texture_size = core::dimension2du(2048, 2048);
+	if (size <= 19) page_texture_size = core::dimension2du(256, 256);
+	else if (size <= 38) page_texture_size = core::dimension2du(512, 512);
+	else if (size <= 76) page_texture_size = core::dimension2du(1024, 1024);
+	else if (size <= 152) page_texture_size = core::dimension2du(2048, 2048);
```

Comment:
How are those semi-magic numbers calculated?

Reply:
largest x that satisfies `(2048/x)**2 >= 180`